dotnet/samples/KernelSyntaxExamples/Example00_01_PlayFabDataQnA.cs

```diff
@@ -434,25 +474,290 @@ private async Task<string> CreateAndExcecutePythonScript(string question, string
     {
-        // Remove from the beginning
-        while (source.StartsWith(wordToRemove, StringComparison.OrdinalIgnoreCase))
+        // Get a reference to the database and container
+        Database database = this._cosmosClient.GetDatabase(this._databaseName);
+        Microsoft.Azure.Cosmos.Container container = database.GetContainer(this._containerName);
+
+        try
+        {
+            // Read the document by its ID
+            ItemResponse<GameReport> response = await container.ReadItemAsync<GameReport>(documentId, new PartitionKey(this._titleId), cancellationToken: cancellationToken);
+            return response.Resource;
+        }
+        catch (CosmosException ex)
         {
-            source = source.Substring(wordToRemove.Length);
+            Console.WriteLine($"Error: {ex.Message}");
+            return null;
         }
+    }
 
-        // Remove from the end
-        while (source.EndsWith(wordToRemove, StringComparison.OrdinalIgnoreCase))
+    public async Task<IList<GameReport>> FetchByQueryAsync(string query, CancellationToken cancellationToken)
+    {
+        // Get a reference to the database and container
+        Database database = this._cosmosClient.GetDatabase(this._databaseName);
+        Microsoft.Azure.Cosmos.Container container = database.GetContainer(this._containerName);
+
+        List<GameReport> ret = new();
+        try
+        {
+            // Execute the query
+            QueryDefinition queryDefinition = new(query);
+            FeedIterator<dynamic> resultSetIterator = container.GetItemQueryIterator<dynamic>(queryDefinition);
+
+            while (resultSetIterator.HasMoreResults)
+            {
+                FeedResponse<dynamic> response = await resultSetIterator.ReadNextAsync(cancellationToken);
+                foreach (dynamic item in response)
+                {
+                    GameReport? gameReport = ((JObject)item).ToObject<GameReport>();
+                    if (gameReport is not null)
+                    {
+                        ret.Add(gameReport);
+                    }
+                }
+            }
+        }
+        catch (CosmosException ex)
         {
-            source = source.Substring(0, source.Length - wordToRemove.Length);
+            Console.WriteLine($"Error: {ex.Message}");
+            return new List<GameReport>();
         }
 
-        return source;
+        return ret;
     }
+
+    public void Dispose()
+    {
+        this._cosmosClient.Dispose();
+    }
+}
+
+public class GameReport
+{
+    public DateTime ReportDate { get; set; }
+    public required string ReportId { get; set; }
+    public required string ReportName { get; set; }
+    public required string ReportData { get; set; }
+}
+
+public class DailyOverviewReportRecord
+{
+    public static string GetHeader() =>
+        "Timestamp,TotalLogins,UniqueLogins,UniquePayers,Revenue,Purchases,TotalCalls,TotalSuccessfulCalls,TotalErrors,Arpu,Arppu,AvgPurchasePrice,NewUsers";
```

Comment:
Do we need these classes even after getting data from cosmos database?

Reply:
Yes, since CosmodDB doesn't give headers or description. It has properties in the non-csv report which are like the properties names of this class